lib/matplotlib/tests/test_image.py

```diff
@@ -117,11 +117,16 @@ def test_image_python_io():
 
 @check_figures_equal()
 def test_imshow_pil(fig_test, fig_ref):
-    pytest.importorskip("PIL")
-    img = plt.imread(os.path.join(os.path.dirname(__file__),
-                     'baseline_images', 'test_image', 'uint16.tif'))
-    fig_test.subplots().imshow(img)
-    fig_ref.subplots().imshow(np.asarray(img))
+    style.use("default")
+    PIL = pytest.importorskip("PIL")
+    png_path = Path(__file__).parent / "baseline_images/pngsuite/basn3p04.png"
+    tiff_path = Path(__file__).parent / "baseline_images/test_image/uint16.tif"
+    axs = fig_test.subplots(2)
+    axs[0].imshow(PIL.Image.open(png_path))
+    axs[1].imshow(PIL.Image.open(tiff_path))
+    axs = fig_ref.subplots(2)
+    axs[0].imshow(plt.imread(str(png_path)))
```

Comment:
Looks good - why is this "str" here?

Reply:
because imread() doesn't support Path inputs *for png only*.